src/dsp_tools/utils/xmlupload/read_validate_xml_file.py

```diff
@@ -9,50 +8,39 @@
 from dsp_tools.models.exceptions import UserError
 from dsp_tools.models.xmlresource import XMLResource
 from dsp_tools.utils.create_logger import get_logger
+from dsp_tools.utils.shared import validate_xml_against_schema
+from dsp_tools.utils.xml_utils import parse_and_clean_xml_file
 
 logger = get_logger(__name__)
 
 
-def parse_xml_file(input_file: Union[str, Path, etree._ElementTree[Any]]) -> etree._Element:
+def validate_and_parse_xml_file(
+    imgdir: str,
+    input_file: Union[str, Path, etree._ElementTree[Any]],
+    preprocessing_done: bool,
+) -> tuple[str, etree._Element, str]:
     """
-    Parse an XML file with DSP-conform data,
-    remove namespace URI from the elements' names,
-    and transform the special tags <annotation>, <region>, and <link>
-    to their technically correct form
-    <resource restype="Annotation">, <resource restype="Region">, and <resource restype="LinkObj">.
+    This function takes an element tree or a path to an XML file
+    It validates the file against the XML schema
+    It checks if all the mentioned bitstream files are in the specified location
+    It retrieves the shortcode and default ontology from the XML file
```

Comment:
It's really necessary to have a correct punctuation, because reStructuredText behaves like markdown: The line breaks are not rendered. So you will see a mess if you hover over this function:
<img width="857" alt="Screenshot 2023-10-04 at 11 26 20" src="https://github.com/dasch-swiss/dsp-tools/assets/39048939/544f14c4-6c6b-4bce-9270-47cb3cbad72c">